py/visdom/server/handlers/socket_handlers.py

```diff
@@ -48,6 +48,24 @@
 # sockets and the poll-based wrappers are using as much shared code as
 # possible. Try to standardize the code between the client-server and
 # visdom-server socket edges.
+
+
+# ============== #
+# About & Naming #
+# ============== #
+
+# 1. *Handler- & *Wrap-classes are intended to have the same functionality
+#   - *Handler (e.g. VisSocketHandler) use WebSockets
+#   - *Wrap (e.g. VisSocketWrap) use polling-based connections instead
+#   - *Wrapper (e.g. VisSocketWrapper) is just a helper class for the respective *Wrap-class 
+#     to process the current state (instead of the state at the time of polling)
+# 2. VisSocket* classes (VisSocketHandler, VisSocketWrap & VisSocketWrapper)
+#   Their goal is to register clients with write access of actual data.
+# 3. Socket* classes (SocketHandler, SocketWrap & SocketWrapper)
+#   Their goal is to register clients with read access of data.
+#   Write access is limited to data and view organization (i.e. layout settings, env removal and env saving)
```

Comment:
Thanks for providing the demystifying details for these :)

Reply:
Yeah, wrapping my head around the code here was a bit tough. I felt that placing my notes here for others is surely helpful.

--

**Maybe here is a good place to ask**:
I am a bit unsure of how big of a PR one would like to have for changes like that, say a refactoring of a large file. ;)

For instance, I do have worked a bit further on that file here; I merged the duplicated code as much as possible and placed the classes in a hierarchy, but felt that the change is altogether too much for one single PR.
Of course, larger code changes need more time to review, but on the other hand it does not make to much sense to split changes up too much, does it?

Reply:
Definitely too much for one PR. I think that a good PR can be large, so long as it's easily summarized. My rule of thumb is the simple sentence summary:
"Splitting up the server into related files, but retaining functionality" makes sense.
"Refactoring socket code to reduce duplicated code" also makes sense.
Putting these two together may be too much.